homeassistant/components/media_player/emby.py

```diff
@@ -96,6 +97,37 @@ def device_update_callback(data):
             _LOGGER.debug("Adding new devices: %s", new_devices)
             async_add_devices(new_devices, update_before_add=True)
 
+        if auto_group:
+            # Compile a list of entity IDs for active emby instances
+            active_entity_ids = []
+
+            for emby_device_id, device in active_emby_devices.items():
+
+                _LOGGER.debug("Indexing active emby device %s", device.name)
+                _LOGGER.debug("Indexing active emby device ID %s", emby_device_id)
+
+                # Reformat device name into the entity ID format used by
+                # HomeAssistant if we don't do this, we won't be able to
+                # create a group of these entities
+                entity_name = device.name.replace(" - ", "__")
+                entity_name = entity_name.replace(" ", "_")
+                # Prepend media_player prefix used by HomeAssistant -
+                # this is how it would appear in the UI
+                entity_name = "media_player." + entity_name
+                # Convert to lower-case for convention purposes
+                entity_name = entity_name.lower()
+                # Add to list
+                active_entity_ids.append(entity_name)
+
+            # Remove the default device from the group
+            active_entity_ids.remove(
+                "media_player.emby__homeassistant__homeassistant")
+
+            # Create the group
+            hass.states.async_set('group.emby', 'off', {
```

Comment:
Creating automatic groups is usually frowned upon. I don't think this is the correct way to do that either. Probably use `set_group` in the group component. But I don't think this will be allowed. @balloob?

Reply:
I was struggling to find an "acceptable" way to do it - I found a forum post saying that wasn't the correct solution, but it also didn't provide a better one. I'm happy to change to a more acceptable method if somebody knows one.

Reply:
The normal and accepted way is that the user defines her own groups. Only base components create groups by default.

Reply:
Some 3rd party integrations create groups by default and then these can be added as regular home assistant entities. But creating home assistant groups should not be done in the platforms. At least that's the usual response from our developers.

Reply:
So, is it acceptable for the user to create an "emby" (or similar group), and then the component adds entities to that group?

Reply:
No, I don't think so.

Reply:
How does the Hue lights component behave in a similar situation? The lights are added to a group, without needing to be defined individually.

Reply:
Home assistant adds the internal Hue group as a light entity. This is ok. Tradfri does the same. It's not ok to create a home assistant group from many light entities in the platform/component.